bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/storeconfigexporter/StoreConfigExporterImpl.java

```diff
@@ -53,31 +55,31 @@
     private static final String STORE_CODE_PROPERTY = "magentoStore";
     private static final String GRAPHQL_ENDPOINT_PROPERTY = "magentoGraphqlEndpoint";
 
-    @Self
-    private SlingHttpServletRequest request;
+    private static final String DEFAULT_GRAPHQL_ENDPOINT = "/api/graphql";
+    private static final String DEFAULT_STORE_VIEW = "default";
 
     @Self(injectionStrategy = InjectionStrategy.OPTIONAL)
     private MagentoGraphqlClient magentoGraphqlClient;
 
-    @Inject
+    @ScriptVariable
     private Page currentPage;
 
-    @Inject
+    @SlingObject
     private Resource resource;
 
-    private String storeView;
-    private String graphqlEndpoint = "/magento/graphql";
+    private String storeView = DEFAULT_STORE_VIEW;
+    private String graphqlEndpoint = DEFAULT_GRAPHQL_ENDPOINT;
     private HttpMethod method = HttpMethod.POST;
     private Page storeRootPage;
-    private Map<String, String> httpHeaders;
+    private Map<String, String> httpHeaders = Collections.emptyMap();
```

Comment:
With that you can remove the null check in L108.

Reply:
good catch. replaced it with an `isEmpty()` to still return a static String instead of instantiating an `ObjectMapper` to generate an empty json object.